src/legacy/core_plugins/input_control_vis/public/control/filter_manager/phrase_filter_manager.ts

```diff
@@ -18,37 +18,36 @@
  */
 
 import _ from 'lodash';
-import { FilterManager } from './filter_manager.js';
-import { esFilters } from '../../../../../../plugins/data/public';
+
+import { FilterManager } from './filter_manager';
+import {
+  esFilters,
+  IFieldType,
+  IndexPattern,
+  FilterManager as QueryFilterManager,
+} from '../../../../../../plugins/data/public';
 
 export class PhraseFilterManager extends FilterManager {
-  constructor(controlId, fieldName, indexPattern, queryFilter) {
+  constructor(
+    controlId: string,
+    fieldName: string,
+    indexPattern: IndexPattern,
+    queryFilter: QueryFilterManager
+  ) {
     super(controlId, fieldName, indexPattern, queryFilter);
   }
 
-  /**
-   * Convert phrases into filter
-   *
-   * @param {array} phrases
-   * @return {object} query filter
-   *   single phrase: match query
-   *   multiple phrases: bool query with should containing list of match_phrase queries
-   */
-  createFilter(phrases) {
-    let newFilter;
+  createFilter(phrases: any): esFilters.PhraseFilter {
+    let newFilter: esFilters.PhraseFilter;
+    // TODO: Fix type to be required
+    const value = this.indexPattern.fields.getByName(this.fieldName) as IFieldType;
```

Comment:
I thing type is dynamically inferred here

Reply:
Ya the method may return undefined and the js was not checking that. I added a check to throw if doesn't exist.

```ts
const value = this.indexPattern.fields.getByName(this.fieldName);

if (!value) {
  throw new Error(`unable to find fieldName: ${this.fieldName} on indexPattern`);
}

if (phrases.length === 1) {
  newFilter = esFilters.buildPhraseFilter(value, phrases[0], this.indexPattern);
} else {
  newFilter = esFilters.buildPhrasesFilter(value, phrases, this.indexPattern);
}
```

Reply:
59aa8a6